packages/amplify-graphql-model-transformer/rds-lambda/handler.ts

```diff
@@ -144,6 +144,36 @@ const getSecretManagerValue = async (secretArn: string | undefined): Promise<{ u
   }
 }
 
+const resolveConnectionStringValue = async (jsonConnectionString: string): Promise<string> => {
+  const parsedJsonConnectionString = JSON.parse(jsonConnectionString);
+  const ssmRequestError = 'Unable to connect to the database. Check the logs for more details.';
+  const ssmLoggedError = 'Unable to fetch the connection Uri from SSM for the provided paths.';
+  console.log(`connection uri SSM paths: ${parsedJsonConnectionString}`);
+  if (Array.isArray(parsedJsonConnectionString)) {
+    for (const connectionUriSsmPath of parsedJsonConnectionString) {
+      console.log(`Trying to fetch connection string from SSM path: ${connectionUriSsmPath}`);
+      try {
+        return await getSSMValue(connectionUriSsmPath);
+      }
+      catch (e) {
+        // try the next secret path;
+        continue;
+      }
+    }
+    console.log(ssmLoggedError);
+    throw new Error(ssmRequestError);
+  }
+  else {
+    try {
+      return await getSSMValue(parsedJsonConnectionString);
+    }
+    catch (e) {
+      console.log(ssmLoggedError);
```

Comment:
These logs may be redundant. IIRC, exceptions are recorded in CW logs.

Reply:
These ones are intentional since the logged message has more details than the error message we send back to the GraphQL client.

Reply:
Did a quick testing, these log statements are redundant. Throwing an error automatically records the full error object as shown in the screenshot.

```ts
  const customErr = 'This is a custom error';
  console.log(customErr);
  throw new Error(customErr);
```
<img width="998" alt="image" src="https://github.com/aws-amplify/amplify-category-api/assets/83315412/e5b1e94b-04fa-4912-bb3d-6ea165b4e79f">

Reply:
That is expected. The situation here is different from above case since the message we log is different from the error we throw which also gets logged but has less information than the additionally added log.
```
const ssmRequestError = 'Unable to connect to the database. Check the logs for more details.';
const ssmLoggedError = 'Unable to fetch the connection Uri from SSM for the provided paths.';

console.log(ssmLoggedError);
throw new Error(ssmRequestError);
```
So the log message is useful since it additionally tells the developer that: We failed to connect to DB because we couldn't **fetch the connection Uri from SSM**.
The reason this isn't put behind `DEBUG_FLAG` setting is because this is a custom error and does not print out any sensitive information other than what we already expect from our public docs and saves the developer time to debug the issue without needing to update the environment variable to debug.